diff --git a/hfsutils/libhfs/volume.c b/hfsutils/libhfs/volume.c
index 37c41d7c3f..8c01b65c14 100644
--- a/hfsutils/libhfs/volume.c
+++ b/hfsutils/libhfs/volume.c
@@ -269,8 +269,8 @@ int v_geometry(hfsvol *vol, int pnum)
 	ERROR(EINVAL, "volume partition is empty");
     }
 
-  if (vol->vlen < 800 * (1024 >> HFS_BLOCKSZ_BITS))
-    ERROR(EINVAL, "volume is smaller than 800K");
+  if (vol->vlen < 8 * (1024 >> HFS_BLOCKSZ_BITS))
+    ERROR(EINVAL, "volume is smaller than 8K");
 
   return 0;
 
